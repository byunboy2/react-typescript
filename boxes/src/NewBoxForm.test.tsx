import { render, fireEvent } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

describe("NewBoxForm component", ()=>{
    it("renders ", () => {
        render(
            <NewBoxForm
            createBox = {()=>{}}
            />
        )
    });
    it("matches the snapshot", function () {
        const { container } = render(
          <NewBoxForm createBox = {()=>{}}/>
        );
        expect(container).toMatchSnapshot();
    });
    it("checks for labels", () => {
        const { container } = render(<NewBoxForm createBox={() => {}} />);
        const labels = ["Height", "Width", "Background Color", "Add a new box!"];

        labels.forEach((label) => {
          expect(container.textContent).toContain(label);
        });
      });

    test('calls createBox with form data when form is submitted', () => {
        const createBox = jest.fn();
        const { getByLabelText, getByText } = render(<NewBoxForm createBox={createBox} />);

        const heightInput = getByLabelText('Height');
        const widthInput = getByLabelText('Width');
        const backgroundColorInput = getByLabelText('Background Color');
        const addBtn = getByText('Add a new box!');

        fireEvent.change(heightInput, { target: { value: '100' } });
        fireEvent.change(widthInput, { target: { value: '200' } });
        fireEvent.change(backgroundColorInput, { target: { value: 'red' } });
        fireEvent.click(addBtn);


        expect(createBox).toHaveBeenCalled();
        expect(createBox).toHaveBeenCalledWith({ height: '100', width: '200', backgroundColor: 'red', id: expect.any(String) });
      });
});
