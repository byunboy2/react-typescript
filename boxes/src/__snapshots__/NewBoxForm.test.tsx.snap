// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBoxForm component matches the snapshot 1`] = `
<div>
  <div>
    <form>
      <div>
        <label
          class="new-box-form"
          for="newBox-height"
        >
          Height
        </label>
        <input
          id="newBox-height"
          name="height"
          value=""
        />
      </div>
      <div>
        <label
          for="newBox-width"
        >
          Width
        </label>
        <input
          id="newBox-width"
          name="width"
          value=""
        />
      </div>
      <div>
        <label
          for="newBox-backgroundColor"
        >
          Background Color
        </label>
        <input
          id="newBox-backgroundColor"
          name="backgroundColor"
          value=""
        />
      </div>
      <button
        class="NewBoxForm-addBtn"
        id="NewBoxForm-addBtn"
      >
        Add a new box!
      </button>
    </form>
  </div>
</div>
`;
